Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -0,0 +1,308 @@
+
+  mTracksPerEvent.reset(new TH1F("tracksperevent", "Matched TRD tracks per Event", 100, 0, 10));
+  getObjectsManager()->startPublishing(mTracksPerEvent.get());
+
+  mTrackletsPerMatchedTrack.reset(new TH1F("trackletspermatchedtrack", "Tracklets per matched TRD track", 100, 0, 10));
+  getObjectsManager()->startPublishing(mTrackletsPerMatchedTrack.get());
+
+  mTriggerPerTF.reset(new TH1F("triggerpertimeframe", "Triggers per TF", 1000, 0, 1000));
+  getObjectsManager()->startPublishing(mTriggerPerTF.get());
+
+  mTriggerWDigitPerTF.reset(new TH1F("triggerwdpertimeframe", "Triggers with Digits per TF", 10, 0, 10));
+  getObjectsManager()->startPublishing(mTriggerWDigitPerTF.get());
+
+  mPulseHeightpro.reset(new TProfile("PulseHeight/mPulseHeightpro", "Pulse height profile  plot;Timebins;Counts", 30, -0.5, 29.5));
+  mPulseHeightpro.get()->Sumw2();
+  getObjectsManager()->startPublishing(mPulseHeightpro.get());
+
+  mPulseHeightperchamber.reset(new TProfile2D("PulseHeight/mPulseHeightperchamber", "mPulseHeightperchamber;Timebin;Chamber", 30, -0.5, 29.5, 540, 0, 540));
+  mPulseHeightperchamber.get()->Sumw2();
+  getObjectsManager()->startPublishing(mPulseHeightperchamber.get());
+  getObjectsManager()->setDefaultDrawOptions(mPulseHeightperchamber.get()->GetName(), "colz");
+}
+
+void PulseHeightTrackMatch::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info) << "initialize TRDDigitQcTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  retrieveCCDBSettings();
+  buildHistograms();
+}
+
+void PulseHeightTrackMatch::startOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info) << "startOfActivity" << ENDM;
+} // set stats/stacs
+
+void PulseHeightTrackMatch::startOfCycle()
+{
+  ILOG(Info) << "startOfCycle" << ENDM;
+}
+
+bool digitIndexCompare_phtm(unsigned int A, unsigned int B, const std::vector<o2::trd::Digit>& originalDigits)
+{
+  // sort into ROC:padrow
+  const o2::trd::Digit *a, *b;
+  a = &originalDigits[A];
+  b = &originalDigits[B];
+  if (a->getDetector() < b->getDetector()) {
+    return 1;
+  }
+  if (a->getDetector() > b->getDetector()) {
+    return 0;
+  }
+  if (a->getPadRow() < b->getPadRow()) {
+    return 1;
+  }
+  if (a->getPadRow() > b->getPadRow()) {
+    return 0;
+  }
+  if (a->getPadCol() < b->getPadCol()) {
+    return 1;
+  }
+  return 0;
+}
+
+void PulseHeightTrackMatch::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  for (auto&& input : ctx.inputs()) {
+    auto digits = ctx.inputs().get<gsl::span<o2::trd::Digit>>("digits");
+    auto tracklets = ctx.inputs().get<gsl::span<o2::trd::Tracklet64>>("tracklets");
+    auto triggerrecords = ctx.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggers");
+    auto tracks = ctx.inputs().get<gsl::span<o2::trd::TrackTRD>>("tracks");
+    auto triggerrecordstracks = ctx.inputs().get<gsl::span<o2::trd::TrackTriggerRecord>>("trigrectrk");
+
+    std::vector<o2::trd::Digit> digitv(digits.begin(), digits.end());
```

Comment:
I do not see the need for `digitv`: it seems to be instantiated as a copy of the digits for one event, then the size is checked, gets sorted, but I do not see where it is used after that.